model-optimizer/extensions/middle/MakeKaldiConstReshapable.py

```diff
@@ -0,0 +1,137 @@
+                                          'value': int64_array([second_dim]),
+                                          'shape': int64_array([1])}).create_node()
+        mem_shape = Concat(graph, {'name': 'gather_memory_shape/' + input_name,
+                                   'axis': 0, 'in_ports_count': 2}).create_node()
+        mem_shape.in_port(0).connect(get_batch.out_port(0))
+        mem_shape.in_port(1).connect(mem_shape_2nd_dim.out_port(0))
+
+    init_value_prev_lstm_output = {}
+    if not mem_shape.out_port(0).disconnected():
+        for dest in mem_shape.out_port(0).get_destinations():
+            if dest.node.soft_get('op') == "Broadcast" and \
+                    dest.node.in_port(1).get_source().node.soft_get('value', []) == np.array([value], dtype=precision):
+                init_value_prev_lstm_output = dest.node
+                break
+
+    if init_value_prev_lstm_output == {}:
+        fill_value = Const(graph, {'name': 'fill_value/'+input_name,
+                                   'value': np.array([value], dtype=precision),
+                                   'shape': int64_array([1])}).create_node()
+        init_value_prev_lstm_output = Broadcast(graph, {'name': 'init_value/'+input_name,
+                                                        }).create_node()
+        init_value_prev_lstm_output.in_port(0).connect(fill_value.out_port(0))
+        init_value_prev_lstm_output.in_port(1).connect(mem_shape.out_port(0))
+
+    return init_value_prev_lstm_output
+
+
+class MakeKaldiConstReshapable(MiddleReplacementPattern):
+    """
+    Add broadcasting of constant nodes based on batch from Parameter node. This approach works only for Kaldi,
+    because it has the same batch in whole graph due to framework specific.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == "kaldi"]
+
+    def run_after(self):
+        from extensions.middle.InsertSelect import AddSelectBeforeMemoryNodePattern
+        from extensions.middle.ReplaceMemoryOffsetWithSplice import ReplaceMemoryOffsetWithMemoryNodePattern
+        from extensions.middle.ReplaceSpliceNodePattern import ReplaceSpliceNodePattern
+        return [AddSelectBeforeMemoryNodePattern, ReplaceMemoryOffsetWithMemoryNodePattern,
+                ReplaceSpliceNodePattern]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        params = graph.get_op_nodes(op="Parameter")
+        batch = params[0].shape[0]
+
+        # check that all Parameters have the same batch
+        for p in params:
+            assert(p.shape[0] == batch,
+                   "Parameter {} have batch different from the {}".format(p.soft_get('name', p.id),
+                                                                          params[0].soft_get('name', params[0].id)))
+
+        # make constants for initialization of ReadValue reshapable
+        reads = graph.get_op_nodes(op='ReadValue')
+        for read in reads:
```

Comment:
nit: merge these two lines into one: `for read in graph.get_op_nodes(op='ReadValue'):`